### ISSUE TITLE
box webbrowser sdk crushed after lunch.

### ISSUE THREAD

**Issue by salexkidd** [box/box-ios-browse-sdk#14]
I heard BoxDev 2015 keynote and Release iOS Web browser SDK.
i tried demo code.

```objective-C

    [BOXContentClient setClientID:@"My Client ID"
                     clientSecret:@"My Client Secret"];

    BOXContentClient *contentClient = [BOXContentClient defaultClient];
    ...
    ...
```
I compled and run code, after crush. log is

```
2015-04-22 12:51:08.073 boxbrowser[20630:768377] -[__NSCFData hasPrefix:]: unrecognized selector sent to instance 0x7fa2f6804a00
2015-04-22 12:51:08.080 boxbrowser[20630:768377] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSCFData hasPrefix:]: unrecognized selector sent to instance 0x7fa2f6804a00'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000010965ec65 __exceptionPreprocess + 165
	1   libobjc.A.dylib                     0x00000001092f7bb7 objc_exception_throw + 45
	2   CoreFoundation                      0x00000001096660ad -[NSObject(NSObject) doesNotRecognizeSelector:] + 205
	3   CoreFoundation                      0x00000001095bc13c ___forwarding___ + 988
	4   CoreFoundation                      0x00000001095bbcd8 _CF_forwarding_prep_0 + 120
	5   boxbrowser                          0x00000001087345d3 +[BOXOAuth2Session usersInKeychain] + 787
	6   boxbrowser                          0x0000000108702c5e +[BOXContentClient users] + 46
	7   boxbrowser                          0x00000001087017fc __33+[BOXContentClient defaultClient]_block_invoke + 60
	8   libdispatch.dylib                   0x000000010d943614 _dispatch_client_callout + 8
	9   libdispatch.dylib                   0x000000010d92f881 dispatch_once_f + 565
	10  boxbrowser                          0x000000010870179d +[BOXContentClient defaultClient] + 141
	11  boxbrowser                          0x00000001086b785b -[ViewController viewDidLoad] + 123
	12  UIKit                               0x0000000109cdb210 -[UIViewController loadViewIfRequired] + 738
	13  UIKit                               0x0000000109cdb40e -[UIViewController view] + 27
	14  UIKit                               0x0000000109bf62c9 -[UIWindow addRootViewControllerViewIfPossible] + 58
	15  UIKit                               0x0000000109bf668f -[UIWindow _setHidden:forced:] + 247
	16  UIKit                               0x0000000109c02e21 -[UIWindow makeKeyAndVisible] + 42
	17  UIKit                               0x0000000109ba6457 -[UIApplication _callInitializationDelegatesForMainScene:transitionContext:] + 2732
	18  UIKit                               0x0000000109ba91de -[UIApplication _runWithMainScene:transitionContext:completion:] + 1349
	19  UIKit                               0x0000000109ba80d5 -[UIApplication workspaceDidEndTransaction:] + 179
	20  FrontBoardServices                  0x00000001102a35e5 __31-[FBSSerialQueue performAsync:]_block_invoke_2 + 21
	21  CoreFoundation                      0x000000010959241c __CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__ + 12
	22  CoreFoundation                      0x0000000109588165 __CFRunLoopDoBlocks + 341
	23  CoreFoundation                      0x0000000109587f25 __CFRunLoopRun + 2389
	24  CoreFoundation                      0x0000000109587366 CFRunLoopRunSpecific + 470
	25  UIKit                               0x0000000109ba7b42 -[UIApplication _run] + 413
	26  UIKit                               0x0000000109baa900 UIApplicationMain + 1282
	27  boxbrowser                          0x00000001086b7b9f main + 111
	28  libdyld.dylib                       0x000000010d977145 start + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```
maybe it is my environment problem. Because, I reset ("Reset content and setting" ) iPhone simulator, then it worked.

**Comment by RicoYao** [comment]
@salexkidd Thanks for this report. Glad it worked once you reset your simulator. Please do let us know if you encounter it again. This crash looks to have happened while the SDK was trying to read from the keychain, so as you say, it might have been caused by an environmental issue.

**Comment by RicoYao** [comment]
The bug causing this was found to be in the Content SDK.
We will apply a fix there: https://github.com/box/box-ios-content-sdk/pull/19

**Comment by bsuvorov** [comment]
Closing since it has been fixed with Content SDK 1.0.2
